integration-tests/ssh/src/test/java/org/apache/camel/quarkus/component/ssh/it/SshTestResource.java

```diff
@@ -45,33 +54,77 @@ public Map<String, String> start() {
             container = new GenericContainer(SSH_IMAGE)
                     .withExposedPorts(SSH_PORT)
                     .withEnv("PASSWORD_ACCESS", "true")
-                    .withEnv("USER_NAME", "test")
-                    .withEnv("USER_PASSWORD", "password")
+                    .withEnv("USER_NAME", USERNAME)
+                    .withEnv("USER_PASSWORD", PASSWORD)
                     .waitingFor(Wait.forListeningPort());
 
             container.start();
 
             LOGGER.info("Started SSH container to {}:{}", container.getHost(),
                     container.getMappedPort(SSH_PORT).toString());
 
-            return CollectionHelper.mapOf(
-                    "quarkus.ssh.host",
-                    container.getHost(),
-                    "quarkus.ssh.port",
-                    container.getMappedPort(SSH_PORT).toString());
+            securedPort = AvailablePortFinder.getNextAvailable();
+
+            var sshd = SshServer.setUpDefaultServer();
+            sshd.setPort(securedPort);
+            sshd.setKeyPairProvider(new FileKeyPairProvider(Paths.get(getHostKey())));
+            sshd.setCommandFactory(new TestEchoCommandFactory());
+            sshd.setPasswordAuthenticator((username, password, session) -> true);
+            sshd.setPublickeyAuthenticator((username, key, session) -> true);
+            sshd.start();
+
+            sshds.add(sshd);
+
+            edPort = AvailablePortFinder.getNextAvailable();
+
+            sshd = SshServer.setUpDefaultServer();
+            sshd.setPort(edPort);
+            sshd.setKeyPairProvider(new FileKeyPairProvider(Paths.get("target/classes/edDSA/key_ed25519.pem")));
+            sshd.setCommandFactory(new TestEchoCommandFactory());
+            sshd.setPasswordAuthenticator((username, password, session) -> true);
+            sshd.setPublickeyAuthenticator((username, key, session) -> true);
+            sshd.start();
+
+            sshds.add(sshd);
+
+            LOGGER.info("Started SSHD server to {}:{}", container.getHost(),
+                    securedPort);
+
+            return Map.of(
+                    "quarkus.ssh.host", "localhost",
+                    "quarkus.ssh.port", container.getMappedPort(SSH_PORT).toString(),
+                    "quarkus.ssh.secured-port", securedPort + "",
+                    "quarkus.ssh.ed-port", edPort + "",
+                    "ssh.username", USERNAME,
+                    "ssh.password", PASSWORD);
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
+    //todo proper path (no target)
```

Comment:
Probably, todo could be removed ? or follow up tickets opened ?

Reply:
removed